### PR TITLE
[dep] upgrades: ROR, Shakapacker, Node, Ruby, misc gem upgrades

### PR DESCRIPTION
**Pull request by rameziophobia**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/593)

## Summary by CodeRabbit

- **Dependencies Updated**
  - Updated Ruby version to 3.3.3 and Node version to 22.3.0 across the project.
  - Updated `react_on_rails` to version 14.0.3 and `shakapacker` to version 8.0.0.
  - Introduced `net-pop` dependency from GitHub.

- **Documentation**
  - Updated setup instructions in README.md to reflect new Node and Ruby versions.

- **Miscellaneous**
  - Removed `node-sass` and `sleep` dependencies from `package.json`.
  - Added `packageManager` field in `package.json`.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -45,10 +45,14 @@ gem "autoprefixer-rails"
...
 
+# Needed until Ruby 3.3.4 is released https://github.com/ruby/ruby/pull/11006
+# Related issue: https://github.com/ruby/net-pop/issues/26
+gem "net-pop", github: "ruby/net-pop"
+
 # FIXME: quick fix for rails6, not needed since rails 7.0.1
 # https://github.com/rails/rails/pull/44083
 gem "net-imap", require: false
-gem "net-pop", require: false
+# gem "net-pop", require: false
```
Comment (justin808) [inline Gemfile:55]: maybe delete?

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-v18.13.0
+v18.20.3
```
Comment (justin808) [inline .nvmrc:1]: why not 22?

File: .controlplane/Dockerfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ RUN apt-get update -qq && \
 
 # Install JavaScript dependencies
 # Make sure NODE_VERSION matches the node version in .nvmrc and package.json
-ARG NODE_VERSION=18.13.0
+ARG NODE_VERSION=18.20.3
```
Comment (justin808) [inline .controlplane/Dockerfile:11]: 22?

**Review by justin808 — CHANGES_REQUESTED** [review]

File: .controlplane/Dockerfile (hunk 3 above)
Comment (rameziophobia) [reply .controlplane/Dockerfile:11]: Hmmm Let's do that. I will upgrade to node v22 and see if there are any issues.

File: Gemfile (hunk 1 above)
Comment (rameziophobia) [reply Gemfile:55]: I thought it was left there as a comment for tutorial purposes if someone is copying parts of the repo. I can just remove that block

**Pull request merged**